distribution/lib/Standard/Base/0.0.0-dev/src/Meta.enso

```diff
@@ -561,3 +568,24 @@ get_short_type_name typ = @Builtin_Method "Meta.get_short_type_name"
    - constructor: the constructor to get the declaring type of.
 get_constructor_declaring_type : Any -> Any
 get_constructor_declaring_type constructor = @Builtin_Method "Meta.get_constructor_declaring_type"
+
+instrumentor_builtin op args = @Builtin_Method "Meta.instrumentor_builtin"
+
+# Builder to create instrumentation for a function
+type Instrumentor
+    Value impl
+    on_enter self (fn : Text -> Any | Nothing) =
```

Comment:
This interface is very close to [GraalVM Insight](https://www.graalvm.org/tools/javadoc/org/graalvm/tools/insight/Insight.html), but it also is __special__ as it deals with `UUID`s - while [GraalVM Insight](https://www.graalvm.org/tools/javadoc/org/graalvm/tools/insight/Insight.html) locates the nodes with:
```java
class OnConfig {
        public boolean expressions;
        public boolean statements;
        public boolean roots;

        /** [String](http://docs.oracle.com/javase/17/docs/api/java/lang/String.html?is-external=true) with a regular expression to match name of functions.
         * Prior to version 0.6 this had to be a
         * {@code Function<[String](http://docs.oracle.com/javase/17/docs/api/java/lang/String.html?is-external=true),[Boolean](http://docs.oracle.com/javase/17/docs/api/java/lang/Boolean.html?is-external=true)>}.
         */
        public [String](http://docs.oracle.com/javase/17/docs/api/java/lang/String.html?is-external=true) rootNameFilter;
        /* @since 0.4 */
        public [Predicate](http://docs.oracle.com/javase/17/docs/api/java/util/function/Predicate.html?is-external=true)<SourceInfo> sourceFilter;
        /**
         * org.graalvm.tools.lsp.server.types.Location in the source file.
         * @since 1.2
         */
        public OnConfigAt at;
    }

    class OnConfigAt {
        /**
         * [String](http://docs.oracle.com/javase/17/docs/api/java/lang/String.html?is-external=true) with a regular expression to match source path.
         * Exactly one of this or {@link #sourceURI} is a mandatory property of
         * the `at` object.
         * @since 1.2
         */
        public [String](http://docs.oracle.com/javase/17/docs/api/java/lang/String.html?is-external=true) sourcePath;
        /**
         * [String](http://docs.oracle.com/javase/17/docs/api/java/lang/String.html?is-external=true) representation of a source URI.
         * Exactly one of this or {@link #sourcePath} is a mandatory property of
         * the `at` object.
         * @since 1.2
         */
        public [String](http://docs.oracle.com/javase/17/docs/api/java/lang/String.html?is-external=true) sourceURI;
        /**
         * The line to match.
         * @since 1.2
         */
        public [Object](http://docs.oracle.com/javase/17/docs/api/java/lang/Object.html?is-external=true) line;
        /**
         * The column to match.
         * @since 1.2
         */
        public [Object](http://docs.oracle.com/javase/17/docs/api/java/lang/Object.html?is-external=true) column;
    }
```
It would be better to reuse the Insight infrastructure as it is well tested and can run _at full speed_. The current implementation the `Instrumentor` is magnitudes slower as it runs behind `@TruffleBoundary`.

Possible idea: [reuse SymbolProvider](https://www.graalvm.org/tools/javadoc/org/graalvm/tools/insight/Insight.SymbolProvider.html) to give Insight scripts access to UUIDs while keeping all the [Insight manual goodies](https://www.graalvm.org/latest/tools/graalvm-insight/manual/).